### ISSUE TITLE
Do you have any plans to support Swagger 2.0?

### ISSUE THREAD

**Issue by lalpert** [conorroche/swagger-doclet#4]

**Comment by conor-roche-rockalltech** [comment]
It was on my radar, but seeing this plugin https://github.com/kongchen/swagger-maven-plugin now supports swagger 2.0 and spring mvc I think it may be time to park this one.

**Comment by kfiles** [comment]
It would be a shame to abandon this doclet, as I really prefer to use javadoc annotations, rather than polluting our code with swagger annotation dependencies.

**Comment by rmpestano** [comment]
Hi @conor-roche-rockalltech, 

can you point the possible changes in the codebase to support swagger 2.0? 

I'm reading the [migration guide](https://github.com/OAI/OpenAPI-Specification/wiki/Swagger-1.2-to-2.0-Migration-Guide) and looks like it's not too hard to migrate or support both specs.

Thanks in advance.

**Comment by eviljava** [comment]
Don't abandon this doclet! Please add support for swagger 2.0 instead!

**Comment by rodney757** [comment]
@eviljava @rmpestano @kfiles @lalpert 

FYI I've started a [fork](https://github.com/rodney757/swagger-doclet) to implement the 2.0 spec. Currently most features are supported.

However all the tests need to be updated. Security Definitions needs to be updated. Authorization support needs to be updated. JSON SubTypes needs to be supported. ExtraApiDeclarations is not currently supported. README.md needs to be updated. And most new features in swagger 2.0 are not supported.

Some of these would be fairly easy to implement. I'll see what I have time for, and I would be happy to accept PR and do a proper release after we add support for what I mentioned above.

**Comment by rmpestano** [comment]
Hi @rodney757, I've migrated to [JaxRS Analyzer ](https://github.com/sdaschner/jaxrs-analyzer) and also blogged about it [here](https://rpestano.wordpress.com/2017/03/01/smooth-and-powerful-swagger-integration-via-jaxrs-analyzer/ ).

**Comment by abstratt** [comment]
@rmpestano I actually found JAX-RS Analyzer before I found this project - and could not use it as its handling of query parameters is fundamentally wrong (they are all considered mandatory) - see sdaschner/jaxrs-analyzer#116.

**Comment by rmpestano** [comment]
yes, that's a limitation but you can workaround that with @DefaultValue("") and then the param will not be mandatory (works for string params).

**Comment by abstratt** [comment]
@rmpestano I had seen that, but then it is not truly based on JAX-RS anymore, right? I have an existing JAX-RS compliant codebase that I wouldn't  want to tweak to work around that bug. Thanks for the idea anyways.
